### PR TITLE
[release-0.53] [vmi-mutator-webhook]: do not inject LimitRange defaults into VMI

### PR DESCRIPTION
**Pull request by enp0s3**
Signed-off-by: enp0s3 [ibezukh@redhat.com](mailto:ibezukh@redhat.com)

This is a manual cherry-pick of https://github.com/kubevirt/kubevirt/pull/9054

**What this PR does / why we need it:**
There were merge conflicts in the e2e tests because of recent refactors in the main branch

**Which issue(s) this PR fixes**
https://bugzilla.redhat.com/show_bug.cgi?id=2152534

**Release note:**

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator_test.go [hunk 1]
```diff
@@ -194,14 +175,12 @@ var _ = Describe("VirtualMachineInstance Mutator", func() {
...
-	})
-
-	It("should apply defaults on VMI create", func() {
+	DescribeTable("should apply defaults on VMI create", func(arch string, cpuModel string) {
 		// no limits wanted on this test, to not copy the limit to requests
-		mutator.NamespaceLimitsInformer, _ = testutils.NewFakeInformerFor(&k8sv1.LimitRange{})
+		defer func() {
+			webhooks.Arch = rt.GOARCH
+		}()
+		webhooks.Arch = arch
```
Comment (jean-edouard) [inline pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator_test.go:183]: Is this hunk necessary? Looks like it belongs to [`a62049a` (#7007)](https://github.com/kubevirt/kubevirt/pull/7007/commits/a62049a358a25d10d55d5e318820fe2dca612455)

File: pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator_test.go [hunk 2]
```diff
@@ -220,9 +199,11 @@ var _ = Describe("VirtualMachineInstance Mutator", func() {
...
 	})
 
-	It("should apply configurable defaults on VMI create", func() {
-		// no limits wanted on this test, to not copy the limit to requests
-		mutator.NamespaceLimitsInformer, _ = testutils.NewFakeInformerFor(&k8sv1.LimitRange{})
+	DescribeTable("should apply configurable defaults on VMI create", func(arch string, cpuModel string) {
+		defer func() {
+			webhooks.Arch = rt.GOARCH
+		}()
+		webhooks.Arch = arch
```
Comment (jean-edouard) [inline pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator_test.go:206]: Same as above

**Review by jean-edouard — COMMENTED** [review]
See in-line for comments.
PR description also needs BZ update.

File: pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator_test.go (hunk 1 above)
Comment (enp0s3) [reply pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator_test.go]: Done. Thanks!

File: pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator_test.go (hunk 2 above)
Comment (enp0s3) [reply pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator_test.go]: Done. Thanks!

**Review by jean-edouard — COMMENTED** [review]
/approve

**Pull request merged**